client/driver/docker.go

```diff
@@ -1351,3 +1337,40 @@ func calculatePercent(newSample, oldSample, newTotal, oldTotal uint64, cores int
 
 	return (float64(numerator) / float64(denom)) * float64(cores) * 100.0
 }
+
+// authOptionsFrom takes the Docker auth config file and the repo being pulled
+// and returns an AuthConfiguration or an error if the file/repo could not be
+// parsed or looked up.
+func authOptionFrom(file, repo string) (*docker.AuthConfiguration, error) {
```

Comment:
I would like to see a test for this. Hard to follow what this method expects to return the right AuthConfiguration.